### PR TITLE
Fix/fix consensus error

### PR DESCRIPTION
**Pull request by jzhe886**
1. fix bloom filter panic.
2. fix leader election results in repeated ack transactions.

### CODE REVIEW

File: internal/plugins/order/etcdraft/node.go [hunk 1]
```diff
@@ -393,11 +407,12 @@ func (n *Node) publishEntries(ents []raftpb.Entry) bool {
 			//        https://github.com/coreos/etcd/pull/7899). In this
 			//        scenario, when the node comes back up, we will re-apply
 			//        a few entries.
-			if n.getBlockAppliedIndex() >= ents[i].Index {
-				// after commit, update appliedIndex
+			blockAppliedIndex := n.getBlockAppliedIndex()
+			if blockAppliedIndex >= ents[i].Index {
```
Comment (Alexader) [inline internal/plugins/order/etcdraft/node.go:411]: Spliting this line has any advantages?

File: internal/plugins/order/etcdraft/txpool/txpool.go [hunk 2]
```diff
@@ -91,31 +102,29 @@ func (tp *TxPool) PoolSize() int {
...
 }
 
-//Remove stored transactions
+//RemoveTxs remove txs from the cache
 func (tp *TxPool) RemoveTxs(hashes []types.Hash, isLeader bool) {
-	if isLeader {
-		tp.BatchDelete(hashes)
-	}
+	tp.Lock()
+	defer tp.Unlock()
```
Comment (Alexader) [inline internal/plugins/order/etcdraft/txpool/txpool.go:108]: Why need to expand lock section？

File: internal/plugins/order/etcdraft/node.go (hunk 1 above)
Comment (jzhe886) [reply internal/plugins/order/etcdraft/node.go:411]: No advantages, just for split the line.

File: internal/plugins/order/etcdraft/txpool/txpool.go (hunk 2 above)
Comment (jzhe886) [reply internal/plugins/order/etcdraft/txpool/txpool.go:108]: BatchDelete will spend more time to delete the transaction from the leveldb store, tp.lock just for locking the pendingtxs list.

**Pull request merged**